dragonfly/implementations/psyllid_provider.py

```diff
@@ -0,0 +1,279 @@
+            return self.status_value_dict[channel]
+
+
+
+    def activate(self, channel):
+        if self.status_value_dict[channel] == 0:
+            logger.info('Activating Psyllid instance for channel {}'.format(channel))
+            result = self.provider.cmd(self.queue_dict[channel], 'activate-daq')
+        else:
+            logger.warning('Cannot activate Psyllid instance of channel {}'.format(channel))
+            return False
+        time.sleep(1)
+        self.request_status(channel)
+        if self.status_value_dict[channel]!=4:
+            logger.warning('Activating failed')
+            return False
+        else: return True
+
+
+    def deactivate(self, channel):
+        if self.status_value_dict[channel] != 0:
+            logger.info('Deactivating Psyllid instance of channel {}'.format(channel))
+            result = self.provider.cmd(self.queue_dict[channel],'deactivate-daq')
+        else:
+            logger.warning('Cannot deactivate Psyllid instance of channel {}'.format(channel))
+            return False
+        time.sleep(1)
+        self.request_status(channel)
+        if self.status_value_dict[channel]!=0:
+            logger.warning('Deactivating failed')
+            return False
+        else: return True
+
+
+    def reactivate(self, channel):
+        if self.status_value_dict[channel] == 4:
+            logger.info('Reactivating Psyllid instance of channel {}'.format(channel))
+            result = self.provider.cmd(self.queue_dict[channel], 'reactivate-daq')
+            logger.info(result)
+        else:
+            logger.warning('Cannot reactivate Psyllid instance of channel {}'.format(channel))
+            return False
+        time.sleep(1)
+        self.request_status(channel)
+        if self.status_value_dict[channel]!=4:
+            logger.warning('Reactivating failed')
+            return False
+        else:
+            return True
+
+
+    def quit_psyllid(self, channel):
+        result = self.provider.cmd(self.queue_dict[channel], 'quit-psyllid')
+        logger.info('psyllid quit!')
+
+
+    @property
+    def all_central_frequencies(self):
+        return self.freq_dict
+
+
+    def get_central_frequency(self, channel):
+        if self.mode_dict[channel] == 'streaming':
+            request = 'active-config.'+self.channel_dict[channel]+'.strw'
+            result = self.provider.get(self.queue_dict[channel]+'.'+request)
+            logger.info('Psyllid says cf is {}'.format(result['center-freq']))
+            self.freq_dict[channel]=result['center-freq']
+            return self.freq_dict[channel]
+        if self.mode_dict[channnel] == 'triggered':
+            request = 'active-config.'+self.channel_dict[channel]+'.trw'
+            result = self.provider.get(self.queue_dict[channel]+'.'+request)
+            logger.info('Psyllid says cf is {}'.format(result['center-freq']))
+            self.freq_dict[channel]=result['center-freq']
+            return self.freq_dict[channel]
```

Comment:
I suggest the following rework to simplify this method (note how the repeated lines are eliminated):
```
def get_central_frequency(self, channel):
    routing_key_map = {
                                     'streaming': '.strw',
                                     'triggered': '.trw',
                                    }
    request = 'active-config.' + self.channel_dict[channel] + routing_key_map[self.mode_dict[channel]]
    result = self.provider.get(self.queue_dict[channel]+'.'+request)
    logger.info('Psyllid says cf is {}'.format(result['center-freq']))
￼    self.freq_dict[channel]=result['center-freq']
￼    return self.freq_dict[channel]
```